src/ConfCrypt/Validation.hs

```diff
@@ -52,29 +52,29 @@ parameterTypesMatchSchema key ConfCryptFile {parameters} =
     where
         decryptAndCompare Parameter {paramName, paramValue, paramType} =
             catchError (runRule paramType paramName =<< decryptValue key paramValue)
-                       (const $ tell ["Error: Could not decrypt " <> paramName])
+                       (pure $ liftIO $ putStrLn ("Error: Could not decrypt " <> T.unpack paramName))
         runRule paramType paramName val =
             case paramType of
                 Nothing -> pure ()
                 Just CInt | all isDigit $ T.unpack val -> pure ()
                 Just CBoolean | T.toLower val == "true" || T.toLower val == "false" -> pure ()
                 Just CString | not (T.null val) -> pure ()
-                Just CString | T.null val -> tell ["Warning: "<> paramName <> " is empty"]
-                Just pt -> tell ["Error: "<> paramName <> " does not match the schema type " <> typeToOutputString pt]
+                Just CString | T.null val -> liftIO $ putStrLn ("Warning: "<> T.unpack paramName <> " is empty")
+                Just pt -> liftIO $ putStrLn ("Error: "<> T.unpack paramName <> " does not match the schema type " <> T.unpack (typeToOutputString pt))
 
 -- | Raise an error if there are parameters without a schema
-logMissingSchemas :: (Monad m, MonadWriter [T.Text] m) =>
+logMissingSchemas :: MonadIO m =>
```

Comment:
All of the `MonadIO` constraints can be removed once `Text` messages are accumulated.

Reply:
Done in last commit.